File: vowel_spellchecker.go
import (
    "strings"
)

func spellchecker(wordlist []string, queries []string) []string {
    exactMatch := make(map[string]string)
    caseInsensitive := make(map[string]string)
    vowelError := make(map[string]string)

    for _, word := range wordlist {
        exactMatch[word] = word
        lowerWord := strings.ToLower(word)
        if _, exists := caseInsensitive[lowerWord]; !exists {
            caseInsensitive[lowerWord] = word
        }
        vowelWord := replaceVowels(lowerWord)
        if _, exists := vowelError[vowelWord]; !exists {
            vowelError[vowelWord] = word
        }
    }

    result := make([]string, len(queries))
    for i, query := range queries {
        if word, exists := exactMatch[query]; exists {
            result[i] = word
        } else if word, exists := caseInsensitive[strings.ToLower(query)]; exists {
            result[i] = word
        } else if word, exists := vowelError[replaceVowels(strings.ToLower(query))]; exists {
            result[i] = word
        } else {
            result[i] = ""
        }
    }

    return result
}

func replaceVowels(word string) string {
    vowels := "aeiou"
    result := []rune(word)
    for i, char := range result {
        if strings.ContainsRune(vowels, char) {
            result[i] = '*'
        }
    }
    return string(result)
}

File: vowelSpellchecker.py
from typing import List

class Solution:
    def spellchecker(self, wordlist: List[str], queries: List[str]) -> List[str]:
        vowels = set('aeiou')

        def devowel(word: str) -> str:
            return ''.join('*' if ch in vowels else ch for ch in word.lower())

        exact = set(wordlist)
        cap_map = {}
        vow_map = {}

        for word in wordlist:
            low = word.lower()
            vow = devowel(word)
            if low not in cap_map:
                cap_map[low] = word
            if vow not in vow_map:
                vow_map[vow] = word

        result = []
        for query in queries:
            if query in exact:
                result.append(query)
            elif query.lower() in cap_map:
                result.append(cap_map[query.lower()])
            elif devowel(query) in vow_map:
                result.append(vow_map[devowel(query)])
            else:
                result.append("")
        return result
